torch/linalg/__init__.py

```diff
@@ -80,96 +87,96 @@
 
 Examples::
 
     >>> x = torch.rand(4, 4)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.mm(x, y)
+    >>> z = x @ y
     >>> z
     tensor([[ 1.0000, -0.0000, -0.0000,  0.0000],
             [ 0.0000,  1.0000,  0.0000,  0.0000],
             [ 0.0000,  0.0000,  1.0000,  0.0000],
             [ 0.0000, -0.0000, -0.0000,  1.0000]])
-    >>> torch.max(torch.abs(z - torch.eye(4))) # Max non-zero
+    >>> torch.dist(z, torch.eye(4))
     tensor(1.1921e-07)
 
     >>> # Batched inverse example
     >>> x = torch.randn(2, 3, 4, 4)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.matmul(x, y)
-    >>> torch.max(torch.abs(z - torch.eye(4).expand_as(x))) # Max non-zero
+    >>> z = x @ y
+    >>> torch.dist(z, torch.eye(4).expand_as(x))
     tensor(1.9073e-06)
 
     >>> x = torch.rand(4, 4, dtype=torch.cdouble)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.mm(x, y)
-    >>> z
-    tensor([[ 1.0000e+00+0.0000e+00j, -1.3878e-16+3.4694e-16j,
-            5.5511e-17-1.1102e-16j,  0.0000e+00-1.6653e-16j],
-            [ 5.5511e-16-1.6653e-16j,  1.0000e+00+6.9389e-17j,
-            2.2204e-16-1.1102e-16j, -2.2204e-16+1.1102e-16j],
-            [ 3.8858e-16-1.2490e-16j,  2.7756e-17+3.4694e-17j,
-            1.0000e+00+0.0000e+00j, -4.4409e-16+5.5511e-17j],
-            [ 4.4409e-16+5.5511e-16j, -3.8858e-16+1.8041e-16j,
-            2.2204e-16+0.0000e+00j,  1.0000e+00-3.4694e-16j]],
-        dtype=torch.complex128)
-    >>> torch.max(torch.abs(z - torch.eye(4, dtype=torch.cdouble))) # Max non-zero
+    >>> z = x @ y
+    >>> torch.dist(z, torch.eye(4, dtype=torch.cdouble))
     tensor(7.5107e-16, dtype=torch.float64)
+
+.. _inverse:
+    https://en.wikipedia.org/wiki/Invertible_matrix
 """)
 
 det = _add_docstr(_linalg.linalg_det, r"""
 linalg.det(input, *, out=None) -> Tensor
 
-Computes the determinant of a square matrix :attr:`input`, or of each square matrix
-in a batched :attr:`input`.
-
-This function supports float, double, cfloat and cdouble dtypes.
+Computes the determinant of a square matrix. Supports batched inputs.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
 
-.. note:: The determinant is computed using LU factorization. LAPACK's `getrf` is used for CPU inputs,
-          and MAGMA's `getrf` is used for CUDA inputs.
+.. note:: This function is computed using :func:`torch.lu`.
+          For CUDA inputs, this function synchronizes that device with the CPU.
 
 .. note:: Backward through `det` internally uses :func:`torch.linalg.svd` when :attr:`input` is not
           invertible. In this case, double backward through `det` will be unstable when :attr:`input`
           doesn't have distinct singular values. See :func:`torch.linalg.svd` for more details.
 
+.. seealso::
+
+        :func:`torch.linalg.slogdet` computes the sign and logarithm of the norm of
```

Comment:
Corrected. Also made more explicit the real / complex case. This one is particularly annoying, because it's not completely obvious that the generalisation of sing / absolute value is the polar decomposition. Asking to know this would be too much for people without a background in complex numbers, so both cases have to be spelled out separately.